backend/linkbox/linkbox.go

```diff
@@ -0,0 +1,768 @@
+	}
+
+	if newObject == (Entity{}) {
+		return nil, fs.ErrorObjectNotFound
+	}
+
+	return &Object{
+		fs:      f,
+		remote:  name,
+		modTime: time.Unix(newObject.Ctime, 0),
+		fullURL: newObject.URL,
+		size:    int64(newObject.Size),
+		id:      newObject.ItemID,
+		pid:     newObject.Pid,
+	}, nil
+}
+
+// Mkdir makes the directory (container, bucket)
+//
+// Shouldn't return an error if it already exists
+func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+	var pdir, name string
+
+	fullPath := path.Join(f.root, dir)
+	if fullPath == "" {
+		return nil
+	}
+
+	fullPath = strings.TrimPrefix(fullPath, "/")
+
+	dirs := strings.Split(fullPath, "/")
+	dirs = append([]string{""}, dirs...)
+
+	for i, dirName := range dirs {
+		pdir = path.Join(pdir, dirName)
+		name = dirs[i+1]
+		pid, err := getIdByDir(f, ctx, pdir)
+		if err != nil {
+			return err
+		}
+
+		//"https://www.linkbox.to/api/open/folder_create"
+		requestURL, _ := url.Parse("https://www.linkbox.to/api/open/folder_create")
+		q := requestURL.Query()
+		q.Set("name", name)
+		q.Set("pid", strconv.Itoa(pid))
+		q.Set("token", f.opt.Token)
+		q.Set("isShare", "0")
+		q.Set("canInvite", "1")
+		q.Set("canShare", "1")
+		q.Set("withBodyImg", "1")
+		q.Set("desc", "")
+
+		requestURL.RawQuery = q.Encode()
+		response := getResponse{}
+
+		err = getUnmarshaledResponse(ctx, requestURL.String(), &response)
+		if err != nil {
+			return fmt.Errorf("err in response")
+		}
+
+		if i+1 == len(dirs)-1 {
+			break
+		}
+
+		// response status 1501 means that directory already exists
+		if response.Status != 1 && response.Status != 1501 {
+			return fmt.Errorf("could not create dir[%s]: %s", dir, response.Message)
+		}
+
+	}
+	return nil
+}
+
+// Rmdir removes the directory (container, bucket) if empty
+//
+// Return an error if it doesn't exist or isn't empty
+func (f *Fs) Rmdir(ctx context.Context, dir string) error {
+	fullPath := path.Join(f.root, dir)
+
+	if fullPath == "" {
+		return fs.ErrorDirNotFound
+	}
+
+	fullPath = strings.TrimPrefix(fullPath, "/")
+	dirIds, err := getIdByDir(f, ctx, fullPath)
+
+	if err != nil {
+		return err
+	}
+
+	//"https://www.linkbox.to/api/open/folder_create"
+	requestURL, _ := url.Parse("https://www.linkbox.to/api/open/folder_del")
+	q := requestURL.Query()
+	q.Set("dirIds", strconv.Itoa(dirIds))
+	q.Set("token", f.opt.Token)
+
+	requestURL.RawQuery = q.Encode()
+
+	response := getResponse{}
+	err = getUnmarshaledResponse(ctx, requestURL.String(), &response)
+
+	if err != nil {
+		return fmt.Errorf("err in response")
+	}
+
+	if response.Status != 1 {
+		return fmt.Errorf("could not remove dir: %s", response.Message)
+	}
+	return nil
+}
+
+func (o *Object) SetModTime(ctx context.Context, modTime time.Time) error {
+	return fs.ErrorCantSetModTime
+}
+
+// Open opens the file for read.  Call Close() on the returned io.ReadCloser
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (io.ReadCloser, error) {
+	downloadURL := o.fullURL
```

Comment:
You've ignored all the options here which means that Seek and Range will do the wrong thing.

Reply:
I have tried to make some tests with seek and everything was working well even without handling offset parameter in Open. Then I watch through the code and realize that issue can occur only if we go through fs/reopen.go code. I'm not sure that this thing does but it looks like it handles error scenarios. So I've added handling for open options (offset, range) that sets appropriate HTTP headers like it's done in http backend.